### PR TITLE
True color support

### PR DESCRIPTION
**Pull request by guicho271828**
true-color (24bit) support, update to cl-colors2, better documentation, better source code organization, testing on more lisps

### CODE REVIEW

**Comment by pnathan** [comment]
hello @guicho271828 ! Thank you for your contribution! It appears that the docker system used for the unit tests is dead. You might have noticed this library hasn't been updated lately. :) 

I am in the middle of a big deadline at work; I can try to get the UT fixed tonight or this weekend.

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,41 @@
...
-#        - lx86cl64 -b --load quicklisp.lisp --eval '(progn (quicklisp-quickstart:install) (quit))'
+language: common-lisp
+sudo: false
+
+addons:
+  apt:
+    packages:
+      - libc6-i386
+      - clisp
+      - openjdk-7-jre
```
Comment (pnathan) [inline .travis.yml]: this doesn't exist.

https://travis-ci.org/pnathan/cl-ansi-text/jobs/624632302?utm_mediumnotification&utm_source=github_status

```
apt-get install failed
The command "sudo -E apt-get -yq --no-install-suggests --no-install-recommends $(travis_apt_get_options) install libc6-i386 clisp openjdk-7-jre" failed and exited with 100 during .
```

File: .travis.yml [hunk 2]
```diff
@@ -1,13 +1,41 @@
...
+
+env:
+  global:
+    - PATH=~/.roswell/bin:$PATH
+    - ROSWELL_INSTALL_DIR=$HOME/.roswell
+  matrix:
```
Comment (pnathan) [inline .travis.yml]: nice

File: .travis.yml [hunk 3]
```diff
@@ -1,13 +1,41 @@
...
+    - LISP=cmucl
+    - LISP=alisp
+
+matrix:
+  allow_failures:
+    - env: LISP=clisp
+    - env: LISP=abcl
+    - env: LISP=ecl
+    - env: LISP=cmucl
+    - env: LISP=alisp
```
Comment (pnathan) [inline .travis.yml]: huh! not familiar with `alisp`

File: cl-ansi-text.asd [hunk 4]
```diff
@@ -1,6 +1,9 @@
 (asdf:defsystem #:cl-ansi-text
-  :depends-on ( #:cl-colors #:alexandria)
-  :components ((:file "cl-ansi-text"))
+  :depends-on ( #:cl-colors2 #:alexandria)
+  :serial t
+  :pathname "src/"
+  :components ((:file "cl-ansi-text")
+               (:file "define-colors"))
   :name "cl-ansi-text"
   :version "1.0"
```
Comment (pnathan) [inline cl-ansi-text.asd]: Bump version to 1.1 please!

File: src/define-colors.lisp [hunk 5]
```diff
@@ -0,0 +1,17 @@
+
+(in-package :cl-ansi-text)
+
+(defmacro define-colors ()
```
Comment (pnathan) [inline src/define-colors.lisp]: why a separate file?

File: src/define-colors.lisp [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  `(progn
+     ,@(map 'list
+            (lambda (color)
+              `(defun ,(intern (symbol-name color)) (string &key (effect :unset) (style :foreground))
+                 ,(format nil "Returns a string decolated in ~(~a~)." color)
```
Comment (pnathan) [inline src/define-colors.lisp]: sp - "decorated"

File: .travis.yml (hunk 1 above)
Comment (pnathan) [reply .travis.yml]: not actually sure what the latest Java ABCL supports is, I know there's been work on it, but I haven't been dialed in lately.

Comment (guicho271828) [reply .travis.yml]: Ah I forgot to remove it. The package existed in the past but seems to have been removed from Travis at some point.

Comment (guicho271828) [reply .travis.yml]: It is not required either.

File: .travis.yml (hunk 3 above)
Comment (guicho271828) [reply .travis.yml]: This uses Allegro CL trial edition

**Comment by pnathan** [comment]
Super cool!

File: src/define-colors.lisp (hunk 5 above)
Comment (guicho271828) [reply src/define-colors.lisp]: I am not a big fan of `eval-when` used just for making the constants evaluated in the compile time. Putting this code in a separate file makes the dependency explicit

**Comment by guicho271828** [comment]
Done, Thank you for the prompt response. Also, added more comments about the SLIME REPL integration.

**Comment by guicho271828** [comment]
ping

hope you are not working in the christmas
(but I do lol)

**Comment by pnathan** [comment]
Awesome! Sorry about the delay. I get distracted. >.>

**Pull request merged**